packages/fxa-settings/src/pages/Index/container.tsx

```diff
@@ -0,0 +1,159 @@
+  };
+  const { queryParamModel, validationError } =
+    useValidatedQueryParams(IndexQueryParams);
+
+  const { prefillEmail, deleteAccountSuccess, hasBounced } =
+    location.state || {};
+
+  const isWebChannelIntegration =
+    integration.isSync() || integration.isDesktopRelay();
+
+  // Query param should take precedence
+  const email = queryParamModel.email || currentAccount()?.email;
+  const shouldRedirectToSignin = email && !prefillEmail;
+
+  useEffect(() => {
+    if (shouldRedirectToSignin) {
+      navigate('/signin', {
+        state: {
+          email,
+        },
+      });
+    }
+  });
+
+  const signUpOrSignInHandler = useCallback(
+    async (email: string) => {
+      try {
+        if (!isEmailValid(email)) {
+          return {
+            error: getErrorWithDefinedErrNo(AuthUiErrors.EMAIL_REQUIRED),
+          };
+        }
+        const { exists, hasLinkedAccount, hasPassword } =
+          await authClient.accountStatusByEmail(email, {
+            thirdPartyAuthStatus: true,
+          });
+        if (!exists) {
+          if (isEmailMask(email)) {
+            return {
+              error: getErrorWithDefinedErrNo(
+                AuthUiErrors.EMAIL_MASK_NEW_ACCOUNT
+              ),
+            };
+            // "@firefox" or "@firefox.com" email addresses are not valid
+            // at this time, therefore block the attempt.
+            // the added 'i' disallows uppercase letters
+          } else if (new RegExp('@firefox(\\.com)?$', 'i').test(email)) {
+            return {
+              error: getErrorWithDefinedErrNo(
+                AuthUiErrors.DIFFERENT_EMAIL_REQUIRED_FIREFOX_DOMAIN
+              ),
+            };
+          }
+          // DNS lookup for MX record
+          await checkEmailDomain(email);
+
+          navigate('/signup', {
+            state: {
+              email,
+              emailStatusChecked: true,
+            },
+          });
+          return { error: null };
+        } else {
+          if (isWebChannelIntegration) {
+            const { ok } = await firefox.fxaCanLinkAccount({ email });
+            if (!ok) {
+              return {
+                error: getErrorWithDefinedErrNo(
+                  AuthUiErrors.USER_CANCELED_LOGIN
+                ),
+              };
+            }
+          }
+
+          navigate('/signin', {
+            state: {
+              email,
+              hasLinkedAccount,
+              hasPassword,
+            },
+          });
+        }
+        return { error: null };
+      } catch (error) {
+        return getHandledError(error);
+      }
+    },
+    [authClient, navigate, isWebChannelIntegration]
+  );
+
+  if (validationError) {
+    // TODO: should we bother to check for this?
+    // currently this is only a validation error for query param 'email', so do nothing?
+    // Look @ in FXA-11297
+  }
+
+  if (isUnsupportedContext(integration.data.context)) {
+    hardNavigate('/update_firefox', {}, true);
```

Comment:
Are we planning to bring `/update_firefox` and `/download_firefox` into fxa-settings? I noticed some odd (and perhaps unnecessary?) validation of the context param.

Reply:
Yes, if we don't have these under "other" routes, I need to make a ticket for them.

> I noticed some odd (and perhaps unnecessary?) validation of the context param.
Right yeah, this is handled in Express for Backbone. I started a Slack thread about it the other day since I didn't see why we chose to handle it at that level.